bundles/framework/announcements/view/AnnouncementsHandler.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { StateHandler, controllerMixin, Messaging } from 'oskari-ui/util';
+import { Message } from 'oskari-ui';
+import { ANNOUNCEMENTS_LOCALSTORAGE } from './Constants';
+
+// Handler for announcements. Handles state and service calls.
+
+const getMessage = (key, args) => <Message messageKey={key} messageArgs={args} bundleKey='announcements' />;
+
+class ViewHandler extends StateHandler {
+    constructor () {
+        super();
+        this.fetchAnnouncements();
+        this.state = {
+            modals: [],
+            announcements: [],
+            checked: false
+        };
+    }
+
+    fetchAnnouncements () {
+        jQuery.ajax({
+            type: 'GET',
+            dataType: 'json',
+            url: Oskari.urls.getRoute('Announcements'),
+            success: (pResp) => {
+                this.updateState({
+                    announcements: pResp.data
+                });
+                this.updateModals();
+            },
+            error: function (jqXHR, textStatus) {
+                Messaging.error(getMessage('messages.getFailed'));
+            }
+        });
+    }
+
+    updateModals () {
+        var modals = [];
+        this.state.announcements.forEach((announcement) => {
+            if (announcement.active && this.showModal(announcement.id)) {
+                // if announcement is active, then show pop-up of the content
+                modals.push(announcement);
+            }
+        });
+        this.updateState({
+            modals: modals
+        });
+    }
+
+    setAnnouncementAsSeen (index, checked, id) {
```

Comment:
`checked` is "Don't show again"? Rename it that way to make this easier to maintain. Also index isn't really needed since you can find the announcement from the list based on the id.